manticore/native/memory.py

```diff
@@ -603,6 +603,7 @@ def __init__(self, maps: Optional[Iterable[Map]] = None, cpu=StubCPU()):
         self.cpu = cpu
         self._page2map: MutableMapping[int, Map] = WeakValueDictionary()  # {page -> ref{MAP}}
         self._recording_stack: List = []
+        self._solver = EventSolver()
```

Comment:
I think we want to either still use `SelectedSolver` here or remove all of the `self.cpu_publish("will_solve"` and `self.cpu_publish("did_solve"` surrounding the solver calls, because `EventSolver` takes care of publishing those events internally:

https://github.com/trailofbits/manticore/blob/fd83be7a9929d3645a3df995e2f925d3fb3b4b7f/manticore/core/state.py#L121-L125

Reply:
I removed the duplicated events. I think we should aim to remove code at each PR :smile: